frontend/packages/react-app/src/components/pages/ExplorePage.tsx

```diff
@@ -1,17 +1,53 @@
-import React, {useCallback, useEffect, useState} from "react"
-import { Web3Provider } from "@ethersproject/providers"
-import { web3Modal } from "../../utils/web3Modal"
-import ExplorePageTemplate from "../templates/ExplorePageTemplate"
-import { GET_TOKENS_BALANCE_USER_HOLDS } from "../../graphql/subgraph"
-import { useLazyQuery } from "@apollo/react-hooks"
-import { ethers } from "ethers"
-import {TokenBalanceUserHolds, UserToken} from "../../interfaces";
+import React, { useCallback, useEffect, useState, useReducer } from "react";
+import { Web3Provider } from "@ethersproject/providers";
+import { web3Modal } from "../../utils/web3Modal";
+import ExplorePageTemplate from "../templates/ExplorePageTemplate";
+import { GET_TOKENS_BALANCE_USER_HOLDS } from "../../graphql/subgraph";
+import { useLazyQuery } from "@apollo/react-hooks";
+import { ethers } from "ethers";
+import { TokenBalanceUserHolds, UserToken } from "../../interfaces";
+
+export const ACTIONS = {
+  OPEN_MODAL: "open_modal",
+  CLOSE_MODAL: "close_modal",
+  SET_TOKENS: "set_tokens",
+};
```

Comment:
The react(redux) typescript guide recommend to use Union type of Typescript like following: 

```javascript
type Action =
    { type: "open_modal" }
| { type: "close_modal" }
| { type: "set_tokens", payload: { data: TokenBalanceUserHolds } }

function reducer(state: ExplorePageState, action: Action) {
  switch (action.type) {
    (...)
    case "set_tokens": // string value is checked with Action type 
    (...)
} 
``` 

This style guide is so helpful.
https://github.com/piotrwitek/react-redux-typescript-guide#--usereducer

Reply:
Thanks!